<launch>
    

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lego_loam)/launch/test.rviz" />

    <!--- TF -->


    <!--- LeGO-LOAM -->    
    <node pkg="lego_loam" type="imageProjection"    name="imageProjection"    output="screen">

       <param name="remove_range"  value="0.5" />
       <param name="groundscanind" type="int" value="5" />
       <param name="segtheta"  value="1.0472" />
       <param name="n_scan" type="int" value="16" />
       <param name="h_scan" type="int" value="1800" />
       <param name="lidarTopics" type="string" value="/velodyne_ns/velodyne_points" />

    </node>

    <node pkg="lego_loam" type="VIOAssociation" name="VIOAssociation" output="screen">

       <param name="vio_att_topic" type="string" value="/vio_data_rigid1/att" />
       <param name="vio_pos_topic" type="string" value="/vio_data_rigid1/pos" />
       <param name="n_scan" type="int" value="16" />
       <param name="h_scan" type="int" value="1800" />

    </node>
    
    <!-- <node pkg="lego_loam" type="VIOplus" name="VIOplus" output="screen">
       <param name="n_scan" type="int" value="16" />
       <param name="h_scan" type="int" value="1800" />
    </node> -->

    <node pkg="lego_loam" type="mapOptmization"     name="mapOptmization"     output="screen">

       <param name="loopflag" type="bool" value="true" />
       <param name="imu_topic" type="string" value="/imu/daa" />
       <param name="mapinterval" type="double" value="0.3" />
       <param name="loop_distance"  value="15" />
       <param name="map_range"  value="500.0" />
       <param name="number_keyframe"  value="90.0" />
       <param name="resolution"  value="0.8" />
       <param name="numberlocalmap" type="int" value="11" />
       <param name="localvoxelflag" type="bool" value="false" />
       <param name="use_2Dprojection" type="bool" value="false" />
       <param name="use_passThrough" type="bool" value="false" />
       <param name="use_passThrough_local" type="bool" value="false" />
       <param name="publish_minZ" value="0.5" />
       <param name="publish_maxZ" value="2.5" />
       <param name="publish_local_minZ" value="0.5" />
       <param name="publish_local_maxZ" value="10" />

    </node>

    <node pkg="lego_loam" type="transformFusion"    name="transformFusion"    output="screen">
    </node>

    
    <!-- <node pkg="lego_loam" type="globaloctomapping" name="globaloctomapping" output="screen">
      <param name="global_2Dcloud_topic" type="string" value="/laser_cloud_surround" />
      <param name="mapping_resolution" value="0.8" />
    </node> -->

</launch>
